crates/fraud_proofs/global_merkle_root/storage/src/update.rs

```diff
@@ -305,6 +305,246 @@ impl TransactionOutputs for Transaction {
+            .storage_as_ref::<Coins>()
+            .get(&utxo_id)
+            .unwrap()
+            .unwrap()
+            .into_owned();
+
+        // Then
+        assert_eq!(*inserted_coin.amount(), output_amount);
+        assert_eq!(*inserted_coin.owner(), output_address);
+    }
+
+    #[test]
+    fn process_output__should_insert_latest_contract_utxo_when_contract_created() {
+        let mut rng = StdRng::seed_from_u64(1337);
+
+        // Given
+        let mut storage: InMemoryStorage<Column> = InMemoryStorage::default();
+        let mut storage_tx = storage.write_transaction();
+        let mut update_tx = storage_tx.update_transaction();
+
+        let tx_pointer = random_tx_pointer(&mut rng);
+        let utxo_id = random_utxo_id(&mut rng);
+        let inputs = vec![];
+
+        let contract_id = random_contract_id(&mut rng);
+        let output = Output::ContractCreated {
+            contract_id,
+            state_root: Bytes32::zeroed(),
+        };
+
+        // When
+        update_tx
+            .process_output(tx_pointer, utxo_id, &inputs, &output)
+            .unwrap();
+
+        storage_tx.commit().unwrap();
+
+        let inserted_contract_utxo = storage
+            .read_transaction()
+            .storage_as_ref::<ContractsLatestUtxo>()
+            .get(&contract_id)
+            .unwrap()
+            .unwrap()
+            .into_owned();
+
+        // Then
+        assert_eq!(inserted_contract_utxo.utxo_id(), &utxo_id);
+    }
+
+    #[test]
+    fn process_output__should_update_latest_contract_utxo_when_interacting_with_contract()
+    {
+        let mut rng = StdRng::seed_from_u64(1337);
+
+        // Given
+        let mut storage: InMemoryStorage<Column> = InMemoryStorage::default();
+        let mut storage_tx = storage.write_transaction();
+        let mut update_tx = storage_tx.update_transaction();
+
+        let tx_pointer = random_tx_pointer(&mut rng);
+        let utxo_id = random_utxo_id(&mut rng);
+
+        let contract_id = random_contract_id(&mut rng);
+        let input_contract = input::contract::Contract {
+            contract_id,
+            ..Default::default()
+        };
+        let inputs = vec![Input::Contract(input_contract)];
+
+        let output_contract = output::contract::Contract {
+            input_index: 0,
+            ..Default::default()
+        };
+
+        let output = Output::Contract(output_contract);
+
+        // When
+        update_tx
+            .process_output(tx_pointer, utxo_id, &inputs, &output)
+            .unwrap();
+
+        storage_tx.commit().unwrap();
+
+        let inserted_contract_utxo = storage
+            .read_transaction()
+            .storage_as_ref::<ContractsLatestUtxo>()
+            .get(&contract_id)
+            .unwrap()
+            .unwrap()
+            .into_owned();
+
+        // Then
+        assert_eq!(inserted_contract_utxo.utxo_id(), &utxo_id);
+    }
+
+    #[test]
+    fn process_input__should_remove_coin() {
```

Comment:
As someone with little context here, it's not obvious to me what the difference between `process_output__should_insert_coin` and `process_input__should_remove_coin` are. It would be nice to explain the conditions for coin being added or removed in the test name so I know what I'm looking for.

Reply:
Sure, I'll add some comments. Just restating here, the difference is when a transaction produces a coin as an output, it should be added to the Coins table, since we have created a coin. When a coin is spent as an input, we remove it because it's no longer in the active UTXO set. I'll figure out a nice way to explain this in the test. Thanks for pointing this out. I'd love for the code to be as understandable as possible with as little context as possible.

Reply:
Done in https://github.com/FuelLabs/fuel-core/pull/2598/commits/92b1d22811b720d3d25be1a24ef2be847ae27ec0

Reply:
Let me know what you think.

Reply:
Gonna get this merged now, but still happy to hear your opinions when you're back. Can make a follow-up if you have any further improvement suggestions.